include/numerics/numeric_vector.h

```diff
@@ -132,6 +132,12 @@ class NumericVector : public ReferenceCountedObject<NumericVector<T>>,
   NumericVector (const NumericVector &) = default;
   NumericVector & operator= (NumericVector &&) = default;
 
+  /**
+   * While this class doesn't manage any memory, the derived class might and
+   * users may be deleting through a pointer to this base class
+   */
+  virtual ~NumericVector() {}
```

Comment:
@roystgnr @jwpeterson Is there anything I'm missing here? Are you guys ok with this addition?

Reply:
I'm trying to use the `= default;` notation for empty special functions. I'm wondering how we never had a virtual destructor from NumericVector before now??? Seems like a good one to backport.

Reply:
switched to `default`

Reply:
I'm actually seriously confused here.  Don't we almost always delete NumericVector subclass objects via a pointer to base?  Don't all our subclasses do heap memory deallocation in their destructors?

Oh, I get how it works now.  ParallelObject has a virtual destructor, so all of its derived classes end up having a virtual destructor, whether they're explicitly declared to or not.  I'm a fan of either this or the "= default" solution to make that more bulletproof, though.  Looks like we originally did have an explicit virtual NumericVector destructor, too, at least in 0.5.0 (first ancient version I had handy).

Reply:
Ah...that makes sense. Yea I was definitely perplexed as well

Reply:
Oof, looks like I just randomly deleted it in the process of updating the special functions in f71e04fd58d0618bd5903b239f66d121e3f57b0b